Lesson_3_HW/src/main/java/GU_Android_1089/Lesson_3/HW/Homework_Practice_GuessWordGame.java

```diff
@@ -0,0 +1,89 @@
+package GU_Android_1089.Lesson_3.HW;
+
+import java.util.Locale;
+import java.util.Scanner;
+
+public class Homework_Practice_GuessWordGame {
+
+    public static void main(String[] args) {
+
+        init();
+
+    }
+
+    private static void init(){
+        String[] words = {"apple", "orange", "lemon", "banana", "apricot", "avocado", "broccoli", "carrot", "cherry",
+                "garlic", "grape", "melon", "leak", "kiwi", "mango", "mushroom", "nut", "olive", "pea", "peanut",
+                "pear", "pepper", "pineapple", "pumpkin", "potato"};
+        int hiddenWordIndex = genIntValue();
+        String hiddenWord = words[hiddenWordIndex];
+        String userWord = getUserWord();
+        CompareWords(hiddenWord, userWord);
+
+    }
+
+    private static int genIntValue (){
+        int randomValue = 0;
+        int maxValue = 24;
+        randomValue = (int)(Math.random()*maxValue);
+        return randomValue;
+    }
+
+    private static String  getUserWord (){
+        Scanner sc = new Scanner(System.in);
+        System.out.print("Введите слово: ");
+        String userWord = sc.nextLine();
+        if (userWord == ""){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

Reply:
> I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
> This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

Thank you for the hint, I didn't know this method.